src/Sylius/Bundle/PromotionBundle/Form/DataTransformer/MoneyIntToLocalizedStringTransformer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PromotionBundle\Form\DataTransformer;
+
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use Symfony\Component\Form\Extension\Core\DataTransformer\MoneyToLocalizedStringTransformer;
+
+final class MoneyIntToLocalizedStringTransformer extends MoneyToLocalizedStringTransformer
```

Comment:
Can you tell me, why do we need this form only in CP context? Why don't we need it for regular promotions, which also contains nested, dynamic, fixed priced based forms?

Reply:
Because in CP context we are using FormEvents and when we are using method `setData()` to update values we should set valid data. Actually, we don't know if new data is valid and if we set some string value there is an exception `TransformationFailedException`. I couldn't find information about how to validate this data in this event `PRE_SUBMIT` before setting it and that's why I decided to not transform this value `if !is_numeric` to avoid this exception and set it to check it in our custom validator.